src/Linters/BaseASTLinter.php

```diff
@@ -76,13 +77,43 @@ public function getPrettyTextForNode(
     foreach ($this->ast->traverseWithParents() as list($node, $parents)) {
       if ($node instanceof $target) {
         $error = $this->getLintErrorForNode($node, $parents);
-        if ($error !== null) {
+
+        if ($error !== null && !$this->isLinterDisabled($node, $error)) {
           yield $error;
         }
       }
     }
   }
 
+  /**
+   * Allow users to disable either all linter errors of this type in a file or
+   * specific cases where a linter is used.
+   **/
+  protected function isLinterDisabled(EditableNode $node, LintError $error): bool {
+    // This class of linter errors might be disabled for the entire file, let's find out.
+    if($this->isLinterDisabledForFile($error->getFile())){
```

Comment:
It would be better to make it so that getLintErrors() calls this and exits before you check any nodes, rather than checking it on every one

Reply:
it looks like you're doing this for some kinds of linters but not others - could you make this happen consistently?